### PR TITLE
enable ipmi raw data if xcatdebugmodel set

### PR DESCRIPTION
**Pull request by zet809**
For feature #2735 

Sample output: 
```
[root@fs1 openbmc_test]# rpower frame45cn02 state
frame45cn02: on

[root@fs1 openbmc_test]# chdef -t site xcatdebugmode=1
1 object definitions have been created or modified.

[root@fs1 openbmc_test]# rpower frame45cn02 state
[ipmi_debug] rpower: state(state), raw_cmd: netfn(0x06=>App), cmd(0x38=>Get Channel Authentication Capabilities), data=[142, 4]
[ipmi_debug] rpower: state(state), raw_cmd: netfn(0x06=>App), cmd(0x3b=>Set session privilege level), data=[4]
[ipmi_debug] rpower: state(state), raw_cmd: netfn(0x06=>App), cmd(0x01=>Get Device ID), data=[]
[ipmi_debug] rpower: state(state), raw_cmd: netfn(0x00=>Chassis), cmd(0x01=>Get Chassis Status), data=[]
frame45cn02: on
[ipmi_debug] rpower: state(state), raw_cmd: netfn(0x06=>App), cmd(0x3c=>Close Session), data=[63, 91, 139, 192]
[root@fs1 openbmc_test]# chdef -t site xcatdebugmode=0
1 object definitions have been created or modified.

[root@fs1 openbmc_test]# rpower frame45cn02 state
frame45cn02: on
```

### CODE REVIEW

**Comment by gurevichmark** [comment]
@zet809 
1. Are you planning to merge this into our code, or is this just an example for reference ? 
2. Can you show what the output would look like ?
3. What would be the steps for enabling this output ? Will it be displayed in the log file or console ?

**Comment by whowutwut** [comment]
@zet809  We need some description in the pull request as to what this pull request is for.   It would be good to show the externals when we turn on and off this command.  Also, the feature it's linking to does not have any description in it yet either.

File: xCAT-server/lib/xcat/plugins/ipmi.pm [hunk 1]
```diff
@@ -8162,6 +8163,7 @@ sub process_request {
     if ($::XCATSITEVALS{ipmitimeout}) { $ipmitimeout = $::XCATSITEVALS{ipmitimeout} }
     if ($::XCATSITEVALS{ipmiretries}) { $ipmitrys = $::XCATSITEVALS{ipmitretries} }
     if ($::XCATSITEVALS{ipmisdrcache}) { $enable_cache = $::XCATSITEVALS{ipmisdrcache} }
+    if ($::XCATSITEVALS{xcatdebugmodel}) { $xcatdebugmodel = $::XCATSITEVALS{xcatdebugmodel} }
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/ipmi.pm]: Why is this one `xcatdebugmodel` , isn't the keyword `xcatdebugmode`, or did you create a new keyword?

File: xCAT-server/lib/xcat/plugins/ipmi.pm [hunk 2]
```diff
@@ -8163,7 +8163,7 @@ sub process_request {
     if ($::XCATSITEVALS{ipmitimeout}) { $ipmitimeout = $::XCATSITEVALS{ipmitimeout} }
     if ($::XCATSITEVALS{ipmiretries}) { $ipmitrys = $::XCATSITEVALS{ipmitretries} }
     if ($::XCATSITEVALS{ipmisdrcache}) { $enable_cache = $::XCATSITEVALS{ipmisdrcache} }
-    if ($::XCATSITEVALS{xcatdebugmodel}) { $xcatdebugmodel = $::XCATSITEVALS{xcatdebugmodel} }
+    if ($::XCATSITEVALS{xcatdebugmode}) { $xcatdebugmodel = $::XCATSITEVALS{xcatdebugmode} }
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/ipmi.pm]: @zet809  You have a bug here.  I can't stress enough that we have to UT even the smallest changes...

**Comment by zet809** [comment]
hi, @gurevichmark and @whowutwut , I had updated the description of both the feature and this pull request, thx!

**Pull request merged**